src/Event.cc

```diff
@@ -48,6 +48,11 @@ void Event::Dispatch(bool no_remote)
 	if ( src == util::detail::SOURCE_BROKER )
 		no_remote = true;
 
+	bool done = PLUGIN_HOOK_WITH_RESULT(HOOK_DISPATCH_EVENT, HookDispatchEvent(this, no_remote),
```

Comment:
My thought here was that it could be surprising if `Dispatch()` ends up using another value for `no_remote` than the one passed to the hook. However, true that the caller provided value gets lost this way. I don't have a clear preference so feel free to change it.

Reply:
It seems fine to me here to pass `no_remote` after the Broker case. Looks like a `no_remote = true` happens only in sufficiently few scenarios (I'm spotting `net_done` and `file_open`, but didn't hunt for long) that if a plugin really needs to dig into details, it can do so. Also, if we turned it around, plugin authors might still have to infer that `no_remote` was about to be flipped due to the Broker case. So I think it's fine.

I did wonder whether the `no_broker` state is something that this (or another) hook should be able to change, but I can't think of uses.